### PR TITLE
SONAR-6632 Fail to index rule if description is more than 32kb

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/search/ws/SearchOptions.java [hunk 1]
```diff
@@ -19,23 +19,21 @@
...
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-
 /**
  * Generic options for search web services
- *
  */
+@Deprecated
 public class SearchOptions {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/search/ws/SearchOptions.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/search/ws/BaseMapping.java [hunk 2]
```diff
@@ -27,14 +27,15 @@
...
 import org.sonar.server.search.IndexUtils;
 import org.sonar.server.search.QueryContext;
 
 /**
- * Mapping of search documents (see {@link org.sonar.server.search.BaseDoc}) to WS JSON responses
+ * Mapping of search documents (see {@link BaseDoc}) to WS JSON responses
  */
 @ServerSide
+@Deprecated
 public abstract class BaseMapping<DOC extends BaseDoc, CTX> {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/search/ws/BaseMapping.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 2 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [DefaultIndexSettings.java#L57](https://github.com/SonarSource/sonarqube/blob/8e471dfb205e249e4c0a49bf6936229fa743a444/server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java#L57): Define a constant instead of duplicating this literal "asciifolding" 3 times. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)
1. MINOR [DefaultIndexSettings.java#L57](https://github.com/SonarSource/sonarqube/blob/8e471dfb205e249e4c0a49bf6936229fa743a444/server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java#L57): Define a constant instead of duplicating this literal "porter_stem" 3 times. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
 import org.sonar.api.rules.RuleType;
 import org.sonar.api.utils.Duration;
 import org.sonar.api.utils.KeyValueFormat;
-import org.sonar.server.search.BaseDoc;
+import org.sonar.server.es.BaseDoc;
 
 public class IssueDoc extends BaseDoc implements Issue {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: Shouldn't the 'index' package be also renamed 'es' ?

File: server/sonar-server/src/test/java/org/sonar/server/es/EsTester.java [hunk 4]
```diff
@@ -145,23 +146,26 @@ public void putDocuments(String index, String type, Class<?> testClass, String..
       }
       bulk.add(new IndexRequest(index, type).source(IOUtils.toString(new FileInputStream(file))));
     }
-    bulk.get();
+    BulkResponse response = bulk.get();
+    assertThat(response.hasFailures()).as(response.buildFailureMessage()).isFalse();
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/es/EsTester.java]: Good improvement !

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**